test/rules/no-shadowed-variable/test.ts.lint

```diff
@@ -215,5 +216,105 @@ function creatorFunction(constructor: MyConstructor, filter: MyFunction): void {
 
 for (let i = 0; i < 3; i++) {
     let i = 34;
-        ~       [Shadowed variable: 'i']
+        ~       [err % ('i')]
+}
+
+(function f1() {
+    let f1;
+        ~~ [err % ('f1')]
+})();
+function f2() {
+    const f2 = 0;
+          ~~ [err % ('f2')]
+}
+(function f3() {
+    var f3;
+        ~~ [err % ('f3')]
+})();
+(function f4(f4) {
+             ~~ [err % ('f4')]
+})();
+let f5;
+(function f5(){})()
+          ~~ [err % ('f5')]
+
+class C<C> {
+        ~ [err % ('C')]
+    C<C>(C: C): C {
+      ~ [err % ('C')]
+         ~ [err % ('C')]
+        type C = C;
+             ~ [err % ('C')]
+        namespace C {
+                  ~ [err % ('C')]
+            interface C<C> {}
```

Comment:
now that we're linting for more than just shadowed _variables_, I'm tempted to change the rule name in the next major release to `no-shadowed-name`...

for now, can we at least distinguish the failure messages? users might get confused if they see "shadowed variable" for an interface name or a type parameter. let's use `Shadowed variable: x` for variables and `Shadowed name: y` for everything else